addons/account/models/account_move.py

```diff
@@ -774,11 +774,7 @@ def _search_default_journal(self):
             journal = self.env['account.journal'].search(domain, limit=1)
 
         if not journal:
-            error_msg = _(
-                "No journal could be found in company %(company_name)s for any of those types: %(journal_types)s",
-                company_name=company.display_name,
-                journal_types=', '.join(journal_types),
-            )
+            error_msg = self.env['account.journal']._build_no_journal_error_msg(company.display_name, ', '.join(journal_types))
```

Comment:
The join() should rather be done in the helper function. journal_types parameter should be a list of strings